ir/src/main/scala/org/scalajs/core/ir/Printers.scala

```diff
@@ -817,6 +808,13 @@ object Printers {
           print("createjsclass[")
           print(cls)
           printRow(captureValues, "](", ", ", ")")
+
+        // Transient
+
+        case Transient(value) =>
+          print("<transient ")
+          print(value.getClass.getSimpleName)
```

Comment:
I think simply `toString` is more appropriate here. If somebody uses a general purpose class inside a `Transient` node, they are doing it wrong anyways (meaning they can give it a reasonable `toString`).

We might even define an interface (or even abstract class) that needs to be implemented specifically to avoid this pitfall.

Reply:
I like the idea of an interface. (an abstract class is a bit too constraining for something custom by design)